samza-api/src/main/java/org/apache/samza/startpoint/StartpointVisitor.java

```diff
@@ -24,40 +24,48 @@
 /**
  * Visitor interface for system consumers to implement to support {@link Startpoint}s.
  */
-public interface StartpointConsumerVisitor {
+public interface StartpointVisitor {
 
   /**
    * Seek to specific offset represented by {@link StartpointSpecific}
    * @param systemStreamPartition The {@link SystemStreamPartition} to seek the offset to.
    * @param startpointSpecific The {@link Startpoint} that represents the specific offset.
    */
-  void register(SystemStreamPartition systemStreamPartition, StartpointSpecific startpointSpecific);
+  void visit(SystemStreamPartition systemStreamPartition, StartpointSpecific startpointSpecific);
```

Comment:
Instead of overloading the "visit" method, could you please have different methods for each type? It makes the calls a little more explicit to make sure the correct visit call is made.
e.g. visitSpecific, visitTimestamp, visitOldest

Reply:
I like being explicit as well, although it sort of contradicts @prateekm 's suggestion in the `StartpointManager`, to refactor `writeStartpoint` and `writeStartpointForTask` into the same overloaded name `writeStartpoint` since the parameter types describe what it does. @prateekm - any guidance on when to overload a method name vs. having different but similar method names for the sake of being explicit on the action?